src/main/java/org/springframework/data/couchbase/core/query/OptionsBuilder.java

```diff
@@ -225,6 +228,30 @@ public static String getScopeFrom(Class<?> domainType) {
 		}
 		return null;
 	}
+	
+	public static DurabilityLevel getDurabilityLevel(Class<?> domainType) {
+		if (domainType == null) {
+			return null;
+		}
+		Document document = AnnotatedElementUtils.findMergedAnnotation(domainType, Document.class);
+		return document.durabilityLevel();
+	}
+
+	public static PersistTo getPersistTo(Class<?> domainType) {
+		if (domainType == null) {
+			return null;
+		}
+		Document document = AnnotatedElementUtils.findMergedAnnotation(domainType, Document.class);
+		return document.persistTo();
+	}
+
+	public static ReplicateTo getReplicateTo(Class<?> domainType) {
+		if (domainType == null) {
+			return null;
+		}
+		Document document = AnnotatedElementUtils.findMergedAnnotation(domainType, Document.class);
+		return document.replicateTo();
+	}
```

Comment:
Just two minor changes in each method :

1) The options cannot be null as they get set if they are not NONE (see buildRemoveOptions) and null is not allowed.

2) it's possible for document to be null, so also return NONE.

Also - to run all the integration tests against a couchbase server, replace src/test/resources/integration.properties with one like this:

```
# Options: containerized, mocked, unmanaged
cluster.type=unmanaged
# Entry point configuration if not managed
# value of hostname and ns_server port
cluster.unmanaged.seed=127.0.0.1:8091
cluster.adminUsername=Administrator
cluster.adminPassword=password
cluster.unmanaged.numReplicas=0
```

and then `mvn integration-test`